### PR TITLE
[PWGHF] modification for efficiency weight

### PR DESCRIPTION
**Pull request by zz951**
1. Add efficiency weight both for Lc and Hadron on taskCorrelation
2. Modification for the method to getdaughters( the old one is not able to run now)

### CODE REVIEW

**Comment by vkucera** [comment]
@zz951 Please compile your branch locally before making a PR.

**Comment by zz951** [comment]
> @zz951 Please compile your branch locally before making a PR.

Hi @vkucera, I have found the reason why it comes the errors (there was an update before I compiled and commit it) , could I commit it again in this PR or new PR?

**Comment by vkucera** [comment]
Please commit a fix in your branch and push it to GitHub. Your PR will get updated automatically.

File: PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx [hunk 1]
```diff
@@ -173,27 +157,41 @@ struct HfCorrelatorLcHadronsSelection {
...
+  Configurable<int> numberEventsMixed{"numberEventsMixed", 5, "number of events mixed in ME process"};
   Configurable<int> applyEfficiency{"applyEfficiency", 1, "Flag for applying Lc efficiency weights"};
-  Configurable<int> filterFlagLc{"filterFlagLc", true, "Flag for applying Collision Filter with Lc"};
-  Configurable<int> filterFlagLcMc{"filterFlagLcMc", false, "Flag for applying Mc Collision Filter with Lc"};
-  Configurable<int> nEventForMixedEvent{"nEventForMixedEvent", 5, "number of event to be mixed"};
   Configurable<float> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<float> yCandGenMax{"yCandGenMax", 0.5, "max. gen. cand. rapidity"};
   Configurable<float> etaTrackMax{"etaTrackMax", 0.8, "max. eta of tracks"};
   Configurable<float> dcaXYTrackMax{"dcaXYTrackMax", 0.0025, "max. DCAxy of tracks"};
   Configurable<float> dcaZTrackMax{"dcaZTrackMax", 0.0025, "max. DCAz of tracks"};
```
Comment (apalasciano) [inline PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx:173]: Same here:
```suggestion
  Configurable<float> dcaZTrackMax{"dcaZTrackMax", 1., "max. DCAz of tracks"};
```

Comment (zz951) [reply PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx]: Ok, thank you!

**Review by apalasciano — APPROVED** [review]

**Pull request merged**